src/components/dialog/DefaultDialog.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <q-dialog v-model="show">
+    <q-card class="q-pa-md">
+      <q-btn icon="fa-solid fa-xmark" class="dialog-icon-close" flat round dense v-close-popup/>
+      <q-card-section class="text-h6 q-mb-xl">
+        <slot name="title"></slot>
+      </q-card-section>
+      <slot></slot>
+    </q-card>
+  </q-dialog>
+</template>
+
+<script setup>
+import {
+  ref,
+  onMounted,
+  onUnmounted,
+} from 'vue';
+import DialogEvent from 'src/composables/events/DialogEvent';
+
+const props = defineProps({
+  dialogKey: String,
+});
+const show = ref(false);
+const dialogEventSubscription = ref(null);
```

Comment:
Doesn't need to be a ref, can be safely initialized before the component is mounted
```
const dialogEventSubscription = DialogEvent.subscribe(onDialogEvent);
```
would need to be moved after onDialogEvent() declaration

Reply:
done